docs/Using-Fleet/Detail-Queries-Summary.md

```diff
@@ -171,21 +171,36 @@ select version, errors, warnings from munki_info;
 - Query:
 
 ```sql
-select
+SELECT
     ia.address,
     id.mac
-from
+FROM
     interface_addresses ia
-    join interface_details id on id.interface = ia.interface
-    join routes r on r.interface = ia.interface
-where
-    r.destination = '0.0.0.0'
-    and r.netmask = 0
-    and r.type = 'gateway'
-    and instr(ia.address, '.') > 0
-order by
-    r.metric asc
-limit 1
+    JOIN interface_details id ON id.interface = ia.interface
+	-- On Unix ia.interface is the name of the interface,
+	-- whereas on Windows ia.interface is the IP of the interface.
+    JOIN routes r ON r.interface = ia.interface
+WHERE
+	-- Destination 0.0.0.0/0 is the default route on route tables.
+    r.destination = '0.0.0.0' AND r.netmask = 0
+	-- Type of route is "gateway" for Unix, "remote" for Windows.
+    AND r.type = 'gateway'
+	-- We are only interested on private IPs (some devices have their Public IP as Primary IP too).
+    AND (
+		-- Private IPv4 addresses.
+		inet_aton(ia.address) IS NOT NULL AND (
+			split(ia.address, '.', 0) = '10'
+			OR (split(ia.address, '.', 0) = '172' AND (CAST(split(ia.address, '.', 1) AS INTEGER) & 0xf0) = 16)
+			OR (split(ia.address, '.', 0) = '192' AND split(ia.address, '.', 1) = '168')
+		)
+		-- Private IPv6 addresses start with 'fc' or 'fd'.
+		OR (inet_aton(ia.address) IS NULL AND regex_match(lower(ia.address), '^f[cd][0-9a-f][0-9a-f]:[0-9a-f:]+', 0) IS NOT NULL)
```

Comment:
This is more for my own understanding, but according to [this](https://en.wikipedia.org/wiki/Unique_local_address) ULAs are in the range `fc00::/7` so shouldn't using `^f[cd]` as the reg-exp suffice here?

Reply:
True. I was a little defensive and wanted to check that it looks like a IPv6 address, but maybe we can assume we are always getting a valid IPv6 string from this column... and just do `^f[cd]` as you propose.

Reply:
Defensive is good, let's keep it that way. I was just curious about it.